test/listmovementcontrol.cpp

```diff
@@ -393,3 +393,102 @@ TEST_CASE("move_page_down() moves down by `list height` lines, and respects `wra
 		}
 	}
 }
+
+TEST_CASE("scroll_halfpage_up() moves up by half of `list height` lines",
+	"[ListMovementControl]")
+{
+	auto list_movement = ListMovementControl<ListStub>(dummyFormName, dummyForm, 0);
+	ListStub& list = list_movement;
+
+	GIVEN("a list of 10 items, where the first item is selected") {
+		list.num_lines = 10;
+		list_movement.set_position(0);
+
+		WHEN("scroll_halfpage_up() is called") {
+			list_movement.scroll_halfpage_up();
+
+			THEN("the first item is still selected") {
+				REQUIRE(list.position == 0);
+				REQUIRE(list.scroll_offset == 0);
+			}
+		}
+	}
+
+	GIVEN("a list of height 5 with 10 items, where the 8th item is selected") {
+		list.num_lines = 10;
+		list.height = 5;
+		// Get list into state where scroll offset is 5 and selected position is 7
+		list_movement.move_page_down(false);
+		list_movement.move_page_down(false);
+		list_movement.move_page_down(false);
+		list_movement.set_position(7);
+		REQUIRE(list.position == 7);
+		REQUIRE(list.scroll_offset == 5);
+
+		WHEN("scroll_halfpage_up() is called repeatedly") {
+			THEN("both the scroll offset and the selected item move up by 3 until at the top") {
+				list_movement.scroll_halfpage_up();
+
+				REQUIRE(list.position == 4);
+				REQUIRE(list.scroll_offset == 2);
+
+				list_movement.scroll_halfpage_up();
+
+				REQUIRE(list.position == 1);
+				REQUIRE(list.scroll_offset == 0);
+
+				list_movement.scroll_halfpage_up();
+
+				REQUIRE(list.position == 0);
+				REQUIRE(list.scroll_offset == 0);
+			}
+		}
+	}
+}
+
+TEST_CASE("scroll_halfpage_down() moves down by `list height` lines, and respects `wrap_scroll`",
```

Comment:
`wrap_scroll` is not checked in this test, and it's actually what I wondered while reading the implementation: shouldn't these new operations respect `wrap-scroll`? Do you have a rationale for leaving it out?

Reply:
I thought it didn't make sense for this functionality, but don't have a good reason for why I thought that.